src/main/scala/edu/berkeley/ce/rockslicing/ClassRegistrator.scala

```diff
@@ -0,0 +1,24 @@
+package edu.berkeley.ce.rockslicing
+
+import com.esotericsoftware.kryo.io.{Input, Output}
+import com.esotericsoftware.kryo.{Kryo, Serializer}
+import com.twitter.chill.SingletonSerializer
+import org.apache.spark.serializer.KryoRegistrator
+
+/**
+  * Kryo serializer registrator for all classes that are to be serialized.
+  */
+class ClassRegistrator extends KryoRegistrator {
+  override def registerClasses(kryo: Kryo) {
+    kryo.register(None.getClass, new SingletonSerializer[AnyRef](None))
+    kryo.register(Nil.getClass, new SingletonSerializer[AnyRef](Nil))
+    kryo.register(Some.getClass, new SingletonSerializer[AnyRef](Some))
+    kryo.register(classOf[scala.collection.mutable.WrappedArray$ofRef])
```

Comment:
Yeah, some profiling probably wouldn't be a bad idea. It's a bit black box to me how it goes about boxing everything and serializing it.